overlord/snapstate/handlers.go

```diff
@@ -4899,8 +4905,11 @@ func (m *SnapManager) doSetupKernelSnap(t *state.Task, _ *tomb.Tomb) error {
 	perfTimings.Save(st)
 
 	// Needed so the old drivers tree can be removed later
-	prevRev := snapSt.Current
-	t.Change().Set("previous-kernel-rev", prevRev)
+	setupTask, err := snapSetupTask(t)
+	if err != nil {
+		return err
+	}
+	setupTask.Set("previous-kernel-rev", snapSt.Current)
```

Comment:
do we have tests here that exercise this with more than just the single task?

Reply:
Added now